src/test/java/teammates/test/cases/browsertests/InstructorFeedbackResultsPageUiTest.java

```diff
@@ -637,7 +637,93 @@ public void testIndicateMissingResponses() {
     // TODO unnecessary coupling of FRComments test here. this should be tested separately.
     private void testFeedbackResponseCommentActions() throws Exception {
 
-        ______TS("Failure case: add empty feedback response comment using comment modal in question view");
+        ______TS("GQR view: Failure case: add empty feedback response comment using comment modal");
+
+        resultsPage = loginToInstructorFeedbackResultsPage("CFResultsUiT.instr", "Open Session");
+        resultsPage.displayByGiverQuestionRecipient();
+        resultsPage.loadResultSectionPanel(1, 2);
+
+        resultsPage.clickCommentModalButton("-4-3-0");
+        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm-4-3-0", "");
+        resultsPage.verifyCommentFormErrorMessage("-4-3-0", Const.StatusMessages.FEEDBACK_RESPONSE_COMMENT_EMPTY);
+        resultsPage.closeCommentModal("-4-3-0");
+
+        ______TS("GQR view: Typical case: add new feedback response comments using comment modal");
+
+        resultsPage.clickCommentModalButton("-4-3-0");
+        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm-4-3-0", "test comment 1");
+        resultsPage.verifyCommentRowContent("-4-3-0-1", "test comment 1", "Teammates Test");
+        resultsPage.isElementPresent(By.id("showResponseCommentAddForm-4-3-0"));
+        resultsPage.verifyContainsElement(By.id("visibility-options-4-3-0-1"));
+        resultsPage.closeCommentModal("-4-3-0");
+
+        resultsPage.clickCommentModalButton("-1-1-1");
+        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm-1-1-1", "test comment 2");
+        resultsPage.verifyCommentRowContent("-1-1-1-2", "test comment 2", "Teammates Test");
+        resultsPage.isElementPresent(By.id("showResponseCommentAddForm-1-1-1"));
+        resultsPage.verifyContainsElement(By.id("visibility-options-1-1-1-2"));
+        resultsPage.closeCommentModal("-1-1-1");
+
+        ______TS("GQR view: Typical case: edit an existing feedback response comment using comment modal");
```

Comment:
Preferably the tests are separated, for add, edit and delete. There are reasons that it's better for tests to be more granular.

1. Since we're promoting TDD, if each test only tests one thing (add, edit eg.) it'll be much easier to keep track of your progress by how many tests pass.

2. I think it's quite rewarding when you make a test pass and you wouldn't get that as often if each test has several verifications.

Does that make sense?

Reply:
Yes, I'll separate these tests.

Reply:
@LuLechuan I have instead separated the tests on the basis of view which uses add, edit delete function since they are common.